commit e47029cdc03b62a0200cde1cbbdbf225b3c4dca5 (refs/changes/74/2774/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-10-11T14:28:12-07:00 (2 years ago)
    
    HEAD-2370 add a rsync-to dev script to sdc-headnode.git

diff --git a/bin/rsync-to b/bin/rsync-to
new file mode 100755
index 00000000..a9a176e9
--- /dev/null
+++ b/bin/rsync-to
@@ -0,0 +1,36 @@
+#!/bin/bash
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+#
+# Copyright (c) 2017, Joyent, Inc.
+#
+
+#
+# Rsync local changes to the given Triton DC headnode, including:
+#
+#       LOCAL           HEADNODE GZ
+#       ./tools         /opt/smartdc
+#       ./scripts       /mnt/usbkey/scripts, /usbkey/scripts
+#
+# This script is intended for a quicker development/test cycle. It should not
+# be used to replace thorough testing before committing changes.
+#
+
+set -o xtrace
+set -o errexit
+
+TOP=$(cd $(dirname $0)/../; pwd)
+HEADNODE=$1
+
+RSYNC_FLAGS=
+#RSYNC_FLAGS="$RSYNC_FLAGS --dry-run"
+
+rsync $RSYNC_FLAGS -av ${TOP}/tools/ $HEADNODE:/opt/smartdc/
+
+ssh $HEADNODE /opt/smartdc/bin/sdc-usbkey mount
+rsync $RSYNC_FLAGS -av --no-o --no-p --no-g ${TOP}/scripts/ $HEADNODE:/mnt/usbkey/scripts/
+rsync $RSYNC_FLAGS -av ${TOP}/scripts/ $HEADNODE:/usbkey/scripts/
